### PR TITLE
Refactor Vue usage

### LINKED ISSUE #35: Refactor vue apps
Use a clear data model that does not require all the forced updates.

### PR DESCRIPTION
**Pull request by kinnarr**
Fixes #35 

This moves the configuration and list of metrics  from the first global `MetricQWebView` to a separate `Store` object with better handling for mutations and Vue reactivity. Every instance of the `MetricQWebView` requires its own store.

### CODE REVIEW

File: js/ui/yaxis-popup.js [hunk 1]
```diff
@@ -107,5 +88,64 @@ Vue.component('yaxis-popup', {
...
+    <div class="modal-footer">
+    <button class="btn btn-primary popup_ok" v-on:click="closePopup">
+    OK
+    </button>
+    </div>
+    </div>
+    </div>
+    </div>
+    </div>`
+}
```
Comment (tilsche) [inline js/ui/yaxis-popup.js:151]: Can we have indentation in templates (automatically)?

File: js/ui/metric-popup.js [hunk 2]
```diff
@@ -62,9 +39,175 @@ Vue.component('metric-popup', {
...
+        }
+      )
+    },
+    closePopup: function (evt) {
+      veil.destroy(evt)
+    },
+    closePopupModal: function (evt) {
+      if (evt.target.getAttribute('role') === 'dialog') {
+        veil.destroy(evt)
+      }
```
Comment (tilsche) [inline js/ui/metric-popup.js:172]: I'm confused by the popup handling.
There's the state in Store.
There's are different close functions, but they don't use the the store.
Then there toggle event.
This looks like plenty of things that can and will go wrong.

File: js/ui/configuration-popup.js [hunk 3]
```diff
@@ -75,14 +76,14 @@ Vue.component('configuration-popup', {
       let newValue = parseFloat(this.uiResolution) + increment
       newValue = this.withinRange(document.getElementById('resolution_input'), newValue)
       this.uiResolution = newValue
-      this.$forceUpdate()
+      // this.$forceUpdate()
```
Comment (tilsche) [inline js/ui/configuration-popup.js:79]: Is there a particular reason why many of those lines are left as comments? If not, just remove.

**Review by tilsche — CHANGES_REQUESTED** [review]
Overall I do like the refactoring. The entry point and the store are pretty clean. I definitely think it's a good improvement. But it also seems we're not quite there yet, especially with the popups. I think with minor changes it's safe to merge and go from here with further iterations.

File: js/ui/metric-popup.js (hunk 2 above)
Comment (kinnarr) [reply js/ui/metric-popup.js:172]: `veil` is unfortunately not only UI-sugar, but also handles popup/veil destruction and sometimes saving. I think this is not the scope of this PR and should be addressed in a new one (see #47)

File: js/ui/yaxis-popup.js (hunk 1 above)
Comment (kinnarr) [reply js/ui/yaxis-popup.js:151]: Automatically? No (see for example https://github.com/vuejs/eslint-plugin-vue/issues/1271). For this PR? sure

**Review by tilsche — APPROVED** [review]

**Pull request merged**